### PR TITLE
Update demographics policy to match business rules

### PR DESCRIPTION
**Pull request by jperk51**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

After some discussions in slack we have decided the original implementation of this policy was correct so we are reverting it to match that decision. 

## Related issue(s)
department-of-veterans-affairs/va-mobile-app#5776

## Testing done

Updated tests to allow the returning of uuids

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by tpharrison — DISMISSED** [review]
LGTM.  Thanks!

**Comment by Tonksthebear** [comment]
LGTM

File: modules/mobile/spec/request/gender_identity_request_spec.rb [hunk 1]
```diff
@@ -37,7 +38,7 @@
         it 'returns a 201' do
           gender_identity = VAProfile::Models::GenderIdentity.new(code: 'F')
 
-          VCR.use_cassette('mobile/va_profile/post_gender_identity_success') do
+          VCR.use_cassette('mobile/va_profile/post_gender_identity_success', erb: { csd: }) do
```
Comment (aherzberg) [inline modules/mobile/spec/request/gender_identity_request_spec.rb:41]: Cool way to customize VCRS. we should be doing more of that.

**Review by aherzberg — APPROVED** [review]
LGTM.

**Review by kpethtel — DISMISSED** [review]
👍

**Review by rileyanderson — APPROVED** [review]
Looks good 🚀

**Pull request merged**